presto-bigquery/src/main/java/io/prestosql/plugin/bigquery/BigQueryTableProperties.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.bigquery;
+
+import com.google.cloud.bigquery.RangePartitioning;
+import com.google.cloud.bigquery.StandardTableDefinition;
+import com.google.cloud.bigquery.TableDefinition;
+import com.google.cloud.bigquery.TimePartitioning;
+import com.google.common.collect.ImmutableList;
+import io.prestosql.spi.session.PropertyMetadata;
+
+import java.util.List;
+import java.util.Optional;
+
+import static io.prestosql.spi.session.PropertyMetadata.stringProperty;
+
+public class BigQueryTableProperties
```

Comment:
If this is a general utility to describe all on the table properties, we should add the [clustering](https://cloud.google.com/bigquery/docs/clustered-tables) as well. Clustering is telling to BigQuery to organized the table data based on one or more columns, and is relevant for partitioned table only.

Reply:
@davidrabinowitz I supposed adding support for clustering property later. Do you think we should add it at once (why)?

Reply:
As the idea is to show the table properties, when users set the table clustering property to affect the query performance. If we want to show the table properties, then showing just the partition field gives a partial result.